cmake/clang_dev_tools.cmake

```diff
@@ -0,0 +1,53 @@
+# Run a set of LLVM/Clang tools on the code base to maintain high
+# code standards and catch potential security vulnerabilities
+#
+# Tools ran:
+# - clang-format
+# - clang-tidy
+# - include-what-you-use
+
+# Get c++ files across all targets
+file(
+  GLOB_RECURSE
+  ALL_SOURCE_FILES
```

Comment:
Do we need to relist all the source directories here?  It seems rather brittle.

Reply:
Didn't find a better way to do this, need to look